File: Festival/src/main/java/Validation/ValidatorArtist.java
package Validation;

import Domain.Artist;
import Validation.Exceptions.ValidatorException;

/**
 * Created by Sebi on 18-Mar-17.
 */
public class ValidatorArtist implements IValidator<Artist> {
    @Override
    public void validate(Artist artist) throws ValidatorException {
        String errors = "";
        if (artist.getName().length() == 0){
            errors += "Artist name must not be null";
        }
        if (errors.length() != 0){
            throw new ValidatorException(errors);
        }
    }
}

File: FestivalC#/Festival/Services/Validation/ValidatorArtist.cs
using Festival.Model;
using Festival.Validation.Exceptions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Validation
{
    public class ValidatorArtist : IValidator<Artist>
    {
        public void validate(Artist artist)
        {
            String errors = "";
            if (artist.Name.Length == 0)
            {
                errors += "Artist name must not be null";
            }
            if (errors.Length != 0)
            {
                throw new ValidatorException(errors);
            }
        }
    }
}
